use "permanent" as canonical symbol for dumbslave

ndeloof opened the pull request:
@Symbol first value is the canonical one as identifier (see javadoc https://github.com/jenkinsci/structs-plugin/blob/master/annotation/src/main/java/org/jenkinsci/Symbol.java#L33)

### Proposed changelog entries

* use "permanent" as canonical symbol for DumbSlave

### Submitter checklist

- [x] no JIRA for minor change
- [x] Changelog entry appropriate for the audience affected by the change
- [x] metadata only, no concrete code to be tested
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

ndeloof committed:
use "agent" as canonical symbol for dumbslave

@Symbol first value is the canonical one as identifier (see javadoc https://github.com/jenkinsci/structs-plugin/blob/master/annotation/src/main/java/org/jenkinsci/Symbol.java#L33)

jglick reviewed (commented):
Amends #3304 but, more to the point, #2293. I am not sure I agree with the original decision to even include `slave` (or now `agent`) as a _possible_ symbol—the Javadoc for `@Symbol` clearly states that it should reflect the _choice_ of extension among the various options, where the extension point itself is implied by the context in which the symbol appears. By this logic, `dumb` is the only legitimate symbol. (Or some such synonym—unfortunately `static` is a keyword and thus not available, but perhaps `fixed`?) The fact that a particular choice is in some sense the “default” is meaningless when selecting a symbol. In most installations, this will be the only choice, and thus the only option offered to, say, IDE completion.

Put another say, suppose you have some config which I will express in Groovy purely for sake of an example:

```groovy
agents {
  <SOMETHING> {
    name: '…'
    remoteFS: '…'
  }
}
```

Using `agent` there does not make sense—of course every entry is an agent. We need to distinguish a dumb agent from some other kind:

```groovy
agents {
  dumb {
    name: '…'
    remoteFS: '…'
  }
  swarm {
    name: '…'
    // (no further configuration for this one)
  }
}
```

ndeloof commented:
actually symbol here is used to identify `nodes`, which in theory also includes `master`.
That being said I agree using `agent` is a poor choice, as this has to compare with things like [kubernetesSlave](https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java). I'd prefer a more specific noun being used, to reflect some classic, user managed node. I guess native english would have more inspiration.
By the way, my initial PR to introduce `agent` as a symbol was only to prevent using `slave`, not to offer a better naming.

jglick commented:
> actually symbol here is used to identify nodes, which in theory also includes master.

Oh? Then how so? Either you are _configuring_ nodes, which is not a list that can possibly include master (see `Jenkins.getNodes()` for example); or you are _selecting_ a node, in which case only the `nodeName` matters, not its implementation.

> I'd prefer a more specific noun being used, to reflect some classic, user managed node.

Yes, me too.

ndeloof commented:
in CasC the target API we are configuring is `Node` to populate `Jenkins#nodes`.
But as I said, my goal here is not to propose a better noun, but to use the expected Symbol value priority. I'm fine if any better noun is suggested to amend this PR.

ndeloof committed:
Use "permanent" as canonical symbol

This better reflect the web UI display name "Permanent Agent" for this Agent flavour
https://github.com/ndeloof/jenkins/blob/agent/core/src/main/resources/hudson/slaves/Messages.properties#L32

ndeloof commented:
@jglick amended my PR to use "permanent" as canonical symbol, to better reflect the web UI label

ndeloof commented:
test failure with "BindException: Address already in use" :-\

oleg-nenashev commented on core/src/main/java/hudson/slaves/DumbSlave.java:
```diff
@@ -65,8 +65,8 @@ public DumbSlave(@Nonnull String name, String remoteFS, ComputerLauncher launche
 
-    @Extension @Symbol({"dumb",
-            "slave", "agent" /*because this is in effect the canonical slave type*/})
+    @Extension @Symbol({"permanent" /*because this is in effect the canonical slave type*/, 
```
"because this is in effect the canonical AGENT type"

ndeloof replied:
in this comment I understand "slave" as "hudson.model.Slave", not at the web UI concept

oleg-nenashev reviewed (approved):
LGTM. Test failure is unrelated

daniel-beck commented:
@ndeloof Please do not use origin branches in this repo for your changes. We have problems with too many stale branches as is, let's not make it worse. Thanks!

ndeloof commented:
@daniel-beck my bad, I created and edited this on github web UI
